CI adjustments

chris579 opened the pull request:
To make it easier for the user to get intercept as one artifact this pr adds artifacts including headers + host addon + host extensions in one zipped container.

dedmen commented:
Binary files only with LFS.
Can you make that two archives?
One with just @intercept with the host dll's and their pdb's and another one with the dev tools (headers, client lib)

jonpas commented:
PDBs should be in dev zip too, for smaller download size for regular users.

dedmen commented:
Yeah not quite sure about the PDB. Didn't feel right to me to have them so far away from the files they actually belong to... 
But I guess right now if something crashes then it's the devs doing that...

chris579 commented:
Including pdbs into the host could definitely be useful for Intercept related crashes, especially for the unstable period. If somebody is downloading the intercept host some more mb wouldn't fear him.

jonpas commented:
Host is for user-end, they won't do debugging. PDBs are actually pretty damn big too compared to the rest.

chris579 commented:
crash dumps can be analyzed even if there are no pdbs present at the time of crash, right? If so, the pdbs could be left out for the host thing

jonpas commented:
No, PDBs are required AFAIK. But your regular user is not going to analyze that, rather send the dump.

chris579 commented:
Then the pdbs could be left out for the host addon

chris579 commented:
Added armake to lfs and splitted up like @dedmen described. I also removed the matrix and hardcoded the builds directly to build both win32 and 64 at the same time to include both into artifacts

chris579 commented:
Build is now working properly and artifact script output is minimized

chris579 commented:
The challenge now is to combine travis build output with appveyor artifacts. AFAIK travis can upload artifacts to aws, this could help to combine all together

dedmen commented:
>Building the projects in this solution one at a time. To enable parallel build, please add the "/m" switch.

Have you tested if the client_lib works properly? Would be nice if the issues we had were fixed.

Ready to merge otherwise. After this I'd make the hotfix release.

chris579 commented:
No had no time for testing and can't do till tomorrow. Can the projects inside be build parallel or are they depending on each other?

dedmen commented:
Even if they are dependent parallel build will probably wait for stuff it needs.

chris579 commented:
Saved us whole 2 minutes

chris579 commented:
Doesn't works for me
Errors I got:

```
Severity	Code	Description	Project	File	Line	Suppression State
Error	LNK2001	unresolved external symbol client_eventhandlers_clear	replay	P:\intercept\replay\src\replay\TimestampPacket.obj	1	
Error	LNK2001	unresolved external symbol client_eventhandlers_clear	replay	P:\intercept\replay\src\replay\server.obj	1	
Error	LNK2001	unresolved external symbol client_eventhandlers_clear	replay	P:\intercept\replay\src\replay\stdafx.obj	1	
Error	LNK2001	unresolved external symbol client_eventhandlers_clear	replay	P:\intercept\replay\src\replay\TcpClient.obj	1	
Error	LNK2001	unresolved external symbol client_eventhandlers_clear	replay	P:\intercept\replay\src\replay\TcpServer.obj	1	
Error	LNK2001	unresolved external symbol client_eventhandlers_clear	replay	P:\intercept\replay\src\replay\Client.obj	1	
Error	LNK2001	unresolved external symbol client_eventhandlers_clear	replay	P:\intercept\replay\src\replay\dllmain.obj	1	
Error	LNK2001	unresolved external symbol client_eventhandlers_clear	replay	P:\intercept\replay\src\replay\PlayerCapture.obj	1	
Error	LNK2001	unresolved external symbol client_eventhandlers_clear	replay	P:\intercept\replay\src\replay\replay.obj	1	
Error	LNK2001	unresolved external symbol "class intercept::types::r_string __cdecl intercept::sqf::name(class intercept::types::object const &)" (?name@sqf@intercept@@YA?AVr_string@types@2@ABVobject@42@@Z)	replay	P:\intercept\replay\src\replay\server.obj	1	
Error	LNK2001	unresolved external symbol "class intercept::types::r_string __cdecl intercept::sqf::group_id(class intercept::types::group const &)" (?group_id@sqf@intercept@@YA?AVr_string@types@2@ABVgroup@42@@Z)	replay	P:\intercept\replay\src\replay\server.obj	1	
Error	LNK2001	unresolved external symbol "void * __cdecl operator new(unsigned int,enum std::align_val_t)" (??2@YAPAXIW4align_val_t@std@@@Z)	replay	P:\intercept\replay\src\replay\intercept_client.lib(world.obj)	1	
Error	LNK2001	unresolved external symbol "void * __cdecl operator new(unsigned int,enum std::align_val_t)" (??2@YAPAXIW4align_val_t@std@@@Z)	replay	P:\intercept\replay\src\replay\intercept_client.lib(core.obj)	1	
Error	LNK2001	unresolved external symbol "void * __cdecl operator new(unsigned int,enum std::align_val_t)" (??2@YAPAXIW4align_val_t@std@@@Z)	replay	P:\intercept\replay\src\replay\intercept_client.lib(client_types.obj)	1	
Error	LNK2001	unresolved external symbol "void * __cdecl operator new(unsigned int,enum std::align_val_t)" (??2@YAPAXIW4align_val_t@std@@@Z)	replay	P:\intercept\replay\src\replay\intercept_client.lib(types.obj)	1	
Error	LNK2001	unresolved external symbol "void * __cdecl operator new(unsigned int,enum std::align_val_t)" (??2@YAPAXIW4align_val_t@std@@@Z)	replay	P:\intercept\replay\src\replay\intercept_client.lib(vehicles.obj)	1	
Error	LNK2001	unresolved external symbol "void * __cdecl operator new(unsigned int,enum std::align_val_t)" (??2@YAPAXIW4align_val_t@std@@@Z)	replay	P:\intercept\replay\src\replay\intercept_client.lib(units.obj)	1	
Error	LNK2001	unresolved external symbol "void * __cdecl operator new(unsigned int,enum std::align_val_t)" (??2@YAPAXIW4align_val_t@std@@@Z)	replay	P:\intercept\replay\src\replay\intercept_client.lib(position.obj)	1	
Error	LNK2001	unresolved external symbol "void __cdecl operator delete(void *,unsigned int,enum std::align_val_t)" (??3@YAXPAXIW4align_val_t@std@@@Z)	replay	P:\intercept\replay\src\replay\intercept_client.lib(common_helpers.obj)	1	
Error	LNK2001	unresolved external symbol "void __cdecl operator delete(void *,unsigned int,enum std::align_val_t)" (??3@YAXPAXIW4align_val_t@std@@@Z)	replay	P:\intercept\replay\src\replay\intercept_client.lib(position.obj)	1	
Error	LNK2001	unresolved external symbol "void __cdecl operator delete(void *,unsigned int,enum std::align_val_t)" (??3@YAXPAXIW4align_val_t@std@@@Z)	replay	P:\intercept\replay\src\replay\intercept_client.lib(world.obj)	1	
Error	LNK2001	unresolved external symbol "void __cdecl operator delete(void *,unsigned int,enum std::align_val_t)" (??3@YAXPAXIW4align_val_t@std@@@Z)	replay	P:\intercept\replay\src\replay\intercept_client.lib(core.obj)	1	
Error	LNK2001	unresolved external symbol "void __cdecl operator delete(void *,unsigned int,enum std::align_val_t)" (??3@YAXPAXIW4align_val_t@std@@@Z)	replay	P:\intercept\replay\src\replay\intercept_client.lib(client_types.obj)	1	
Error	LNK2001	unresolved external symbol "void __cdecl operator delete(void *,unsigned int,enum std::align_val_t)" (??3@YAXPAXIW4align_val_t@std@@@Z)	replay	P:\intercept\replay\src\replay\intercept_client.lib(types.obj)	1	
Error	LNK2001	unresolved external symbol "void __cdecl operator delete(void *,unsigned int,enum std::align_val_t)" (??3@YAXPAXIW4align_val_t@std@@@Z)	replay	P:\intercept\replay\src\replay\intercept_client.lib(multiplayer.obj)	1	
Error	LNK2001	unresolved external symbol "void __cdecl operator delete(void *,unsigned int,enum std::align_val_t)" (??3@YAXPAXIW4align_val_t@std@@@Z)	replay	P:\intercept\replay\src\replay\intercept_client.lib(vehicles.obj)	1	
Error	LNK2001	unresolved external symbol "void __cdecl operator delete(void *,unsigned int,enum std::align_val_t)" (??3@YAXPAXIW4align_val_t@std@@@Z)	replay	P:\intercept\replay\src\replay\intercept_client.lib(units.obj)	1	
Error	LNK1120	5 unresolved externals	replay	P:\intercept\replay\src\@replay\intercept\replay.dll	1	
```

dedmen commented:
client eventhandlers fix is already in my pipeline. The two intercept functions are because engine type..
Google says the rest might be crt lib not being linked... But why and how... Grrr...

I'm ready to merge when you are. Your test project is /MT right?

chris579 commented:
> Your test project is /MT right?

Yep, it is. Used my replay project which works fine with my self compiled build.

> I'm ready to merge when you are.

Let's go for it. The errors occurring in this build should be handled in a separate issue/pr as this only aims at improving the artifacts generated. If there is already an issue I will expand to it otherwise I will open a new one.

jonpas commented:
Can we get rid of old build matrix that's commented out?

chris579 commented:
Yep, it's gone now

dedmen commented on appveyor.yml:
```diff
@@ -1,6 +1,6 @@
 version: "{build}"
 
-image: Visual Studio 2017 Preview
+image: Visual Studio 2017
```
I don't remember why I moved to Preview.. hmm..

chris579 replied:
Maybe because of some errors in the stable build. Has already been in August so I assume they are fixed now. Beside of the common errors in the ci build I can't detect any difference.

chris579 commented:
Tested the latest build locally and it works fine. When merged I will do a new PR trying to find the source why the ci build libraries are broken.

chris579 commented:
Only thing I were not able to test is the tag that should be included in the artifact zip containers. We will see next release if this worked.
